.github/workflows/python-publish.yml

```diff
@@ -23,6 +22,9 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install setuptools wheel twine
+    - name: Run security and version checks
+      run: |
+        python release.py stable ${{ github.event.inputs.version }}  # check matching stable versions
```

Comment:
If this is a new release, then the all the versions should be stable. If they are not, they shouldn't be published to the main package.

Reply:
The `stable` argument simply tells the release.py script to match the input version with that found in setup.py, **not** setup-nightly.py. It does not allow any non-stable package to be published since even if the input version was for a nightly package, it would not match the stable version in setup.py.